File: sdks/java-agent/src/main/java/com/carhire/sdk/clients/LocationsClient.java
package com.carhire.sdk.clients;

import com.carhire.sdk.Config;
import com.carhire.sdk.transport.TransportInterface;

import java.util.concurrent.CompletableFuture;

public class LocationsClient {
    private final TransportInterface transport;
    private final Config config;

    public LocationsClient(TransportInterface transport, Config config) {
        this.transport = transport;
        this.config = config;
    }

    public CompletableFuture<Boolean> isSupported(String agreementRef, String locode) {
        return transport.isLocationSupported(agreementRef, locode);
    }
}

File: sdks/nodejs-agent/src/clients/LocationsClient.ts
import { Config } from '../Config';
import { TransportInterface } from '../transport/TransportInterface';

export class LocationsClient {
  constructor(
    private readonly transport: TransportInterface,
    private readonly config: Config
  ) {}

  public async isSupported(agreementRef: string, locode: string): Promise<boolean> {
    return this.transport.isLocationSupported(agreementRef, locode);
  }
}

